kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutputCell.js

```diff
@@ -25,12 +33,43 @@ define([
             showMenu: true,
             lazyRender: true // used in init()
         },
+        isRendered: false,
         OUTPUT_ERROR_WIDGET: 'kbaseNarrativeError',
+        headerShown: false,
+
+        initMetadata: function () {
+            var baseMeta = {
+                kbase: {
+                    dataCell: {}
+                }
+            };
+            if (!this.cell) {
+                return baseMeta;
+            }
+            else {
+                var metadata = this.cell.metadata;
+                if (!metadata || !metadata.kbase) {
+                    metadata = baseMeta;
+                    this.cell.metadata = metadata;
+                }
+                if (!metadata.kbase.dataCell) {
+                    metadata.kbase.dataCell = {};
+                }
+                return metadata;
+            }
+        },
+
         init: function (options) {
+            // handle where options.widget == null.
+            options.widget = options.widget || this.options.widget;
+            // handle where options.widget == 'null' string. I know. It happens.
+            options.widget = options.widget === 'null' ? this.options.widget : options.widget;
```

Comment:
Before coming to CSHL, I worked at Edison Schools and we imported school data from their SIS systems into our servers every night. One night, it broke at a school. I dug into it. Turns out they had a new kid whose name was "Brandon Null", and our importer failed because it said that his last name was a required field.

So our CSR called up the school and said he was missing his last name. They replied, "What are you talking about? His name is Brandon Null. That didn't go through? Why not?" Poor kid. He's gonna have all sorts of problems in life.

The quick fix was when I realized that the importer only failed on insert, not update - so I just change his name to "Brandon NotNull", imported it, and then went into the database to manually change back to "Brandon Null". After that, his record would update without issue.

We never encountered that specific bug again during my tenure.

Reply:
Ha! Well, all over the place in various app specs, there's the JSON value `null` mixed in with the string `"null"`.

Makes for some fun trickiness there, too.